lxd/device/disk.go

```diff
@@ -177,26 +177,167 @@ func (d *disk) validateConfig(instConf instance.ConfigReader) error {
+		"limits.max": validate.IsAny,
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=size)
+		// This option is supported only for the rootfs (`/`).
+		//
+		// Specify a value in bytes (various suffixes supported, see {ref}`instances-limit-units`).
+		// ---
+		//  type: string
+		//  required: no
+		//  shortdesc: Disk size
+		"size": validate.Optional(validate.IsSize),
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=size.state)
+		// This option is similar to {config:option}`device-disk-device-conf:size`, but applies to the file-system volume used for saving the runtime state in VMs.
+		// ---
+		//  type: string
+		//  required: no
+		//  condition: virtual machine
+		//  shortdesc: Size of the file-system volume used for saving runtime state
+		"size.state": validate.Optional(validate.IsSize),
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=pool)
+		//
+		// ---
+		//  type: string
+		//  required: no
+		//  condition: storage volumes managed by LXD
+		//  shortdesc: Storage pool to which the disk device belongs
+		"pool": validate.IsAny,
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=propagation)
+		// Possible values are `private` (the default), `shared`, `slave`, `unbindable`, `rshared`, `rslave`, `runbindable`, `rprivate`.
+		// See the Linux Kernel [shared subtree](https://www.kernel.org/doc/Documentation/filesystems/sharedsubtree.txt) documentation for a full explanation.
+		//
+		// ---
+		//  type: string
+		//  defaultdesc: `private`
+		//  required: no
+		//  shortdesc: How a bind-mount is shared between the instance and the host
+		"propagation": validatePropagation,
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=raw.mount.options)
+		//
+		// ---
+		//  type: string
+		//  required: no
+		//  shortdesc: File system specific mount options
 		"raw.mount.options": validate.IsAny,
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=ceph.cluster_name)
+		//
+		// ---
+		//  type: string
+		//  defaultdesc: `ceph`
+		//  required: for Ceph or CephFS sources
+		//  shortdesc: Cluster name of the Ceph cluster
 		"ceph.cluster_name": validate.IsAny,
-		"ceph.user_name":    validate.IsAny,
-		"boot.priority":     validate.Optional(validate.IsUint32),
-		"path":              validate.IsAny,
-		"io.cache":          validate.Optional(validate.IsOneOf("none", "writeback", "unsafe")),
-		"io.bus":            validate.Optional(validate.IsOneOf("virtio-scsi", "nvme")),
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=ceph.user_name)
+		//
+		// ---
+		//  type: string
+		//  defaultdesc: `admin`
+		//  required: for Ceph or CephFS sources
+		//  shortdesc: User name of the Ceph cluster
+		"ceph.user_name": validate.IsAny,
+		// lxdmeta:generate(entities=device-disk; group=device-conf; key=boot.priority)
+		// A higher value for this option means that the VM boots first.
```

Comment:
Is this here actually true? Putting boot priority on a disk device makes the VM boot first?

Reply:
My understanding of this option is that in the VM boot sequence, when the BIOS/UEFI scans available storage devices for a bootable operating system to load, the higher the value of `boot.priority` for a device, the earlier it will appear in the scan. @tomponline is that right ?

Reply:
The boot.priority setting specifies which order the VM will try and boot from the devices, so yes specifying a higher value will make the VM try and boot from that disk first. This is used when adding a disk device that is backed by and ISO or ISO volume to make it boot from that one first.

Reply:
@ru-fu what do you think about : `A higher value indicates a higher boot precedence, useful for prioritizing boot sources like ISO-backed disks.` ?

Reply:
OK, now I understand. So it's not about which of the VMs boots first, but what the one VM boots from.
Let's make it even clearer:

```
A higher value indicates a higher boot precedence for the disk device.
This is useful for prioritizing boot sources like ISO-backed disks.
```

Reply:
fixed